### PR TITLE
docs: Add docs for `ALTER ROLE ... SET ...` and `ALTER SYSTEM SET` for session variables

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR adds docs for the `ALTER ROLE ... SET ...` feature and documents that `ALTER SYSTEM SET ...` can be used to change the defaults for session variables.

### Motivation

Fixes https://github.com/MaterializeInc/materialize/issues/22107

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Updates user documentation for the `ALTER ROLE ... SET ...` feature, and documents the session defaults can be modified via `ALTER SYSTEM SET ...`

### CODE REVIEW

File: doc/user/layouts/shortcodes/session-variables.html [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
-sql_safe_updates                            | `false`                   | Boolean flag indicating whether to prohibit SQL statements that may be overly destructive.
-standard_conforming_strings                 | `true`                    | Boolean flag indicating whether ordinary string literals (`'...'`) should treat backslashes literally. The only supported value is `true`.
-statement_timeout                           | `10 seconds`              | The maximum allowed duration of `INSERT`, `UPDATE`, and `DELETE` operations. If this value is specified without units, it is taken as milliseconds.
-timezone                                    | `UTC`                     | The time zone for displaying and interpreting timestamps. The only supported value is `UTC`.
+Name                                        | Default value             |  Description                                                                                                                                                           | Override default with [`ALTER SYSTEM SET`](../alter-system-set)
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/session-variables.html:13]:
```suggestion
Name                                        | Default value             |  Description                                                                                                                                                           | [Settable](../alter-system-set)
```

File: doc/user/layouts/shortcodes/system-variables.html [hunk 2]
```diff
@@ -1,7 +1,12 @@
...
 
-Materialize reserves system variables for region-wide configuration. Although it's possible to `SHOW` system variables, you must [contact us](https://materialize.com/contact/) to set the value of most variables (e.g. increasing the maximum number of AWS PrivateLink connections in your region).
-Some system variables can be modified by any superuser (i.e. Frontegg admin).
+Materialize reserves system variables for region-wide configuration. Although it's possible to
+`SHOW` system variables, you must [contact us](https://materialize.com/contact/) to set the value
+of most variables (e.g. increasing the maximum number of AWS PrivateLink connections in your
+region), but some system variables can be modified by any superuser (i.e. Frontegg admin).
+
+A subset of [session variables](../set/#session-variables) can also have their region-wide default
+changed, using `ALTER SYSTEM SET`.
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/system-variables.html:9]:
```suggestion
Materialize reserves system variables for region-wide configuration. You must [contact us](https://materialize.com/contact/) to set the value
of _most_ variables to a different value than the provided default. System variables that can be modified by a _superuser_ (i.e. [`Organization Admin`](/manage/access-control/#role-based-access-control-rbac)) are marked as `Settable` below.
```

File: doc/user/layouts/shortcodes/system-variables.html [hunk 3]
```diff
@@ -1,7 +1,12 @@
...
 
 Name                                        | Default value                                                         | Description                                                                                                            | Settable by superuser |
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/system-variables.html:11]:
```suggestion
Name                                        | Default value                                                         | Description                                                                                                            | Settable |
```

**Review by morsapaes — CHANGES_REQUESTED** [review]

**Comment by morsapaes** [comment]
Thanks, @ParkMyCar!

File: doc/user/layouts/shortcodes/system-variables.html [hunk 4]
```diff
@@ -1,7 +1,12 @@
 ## System variables
 
-Materialize reserves system variables for region-wide configuration. Although it's possible to `SHOW` system variables, you must [contact us](https://materialize.com/contact/) to set the value of most variables (e.g. increasing the maximum number of AWS PrivateLink connections in your region).
-Some system variables can be modified by any superuser (i.e. Frontegg admin).
+Materialize reserves system variables for region-wide configuration. Although it's possible to
+`SHOW` system variables, you must [contact us](https://materialize.com/contact/) to set the value
+of most variables (e.g. increasing the maximum number of AWS PrivateLink connections in your
+region), but some system variables can be modified by any superuser (i.e. Frontegg admin).
+
+A subset of [session variables](../set/#session-variables) can also have their region-wide default
```
Comment (morsapaes) [inline doc/user/layouts/shortcodes/system-variables.html:8]: I find this confusing: you can set the default for _session_ variables using `ALTER SYSTEM`? Tried it out as an admin, and got:

```sql
ALTER SYSTEM SET cluster = 'other_cluster';

Error: permission denied to alter system
Detail: You must be the 'mz_system' role
```

**Comment by morsapaes** [comment]
@jkosh44, just double-checking: am I supposed to be able to `ALTER ROLE` any role as a non-admin user? It seems a bit scary, and AFAIU you can only do this in PostgreSQL as a non-superuser if you have `CREATEROLE` privileges _and_ `ADMIN OPTION` privileges on the role you're trying to alter.

**Comment by jkosh44** [comment]
> @jkosh44, just double-checking: am I supposed to be able to `ALTER ROLE` any role as a non-admin user? It seems a bit scary, and AFAIU you can only do this in PostgreSQL as a non-superuser if you have `CREATEROLE` privileges _and_ `ADMIN OPTION` privileges on the role you're trying to alter.

Any role with the `CREATEROLE` system privilege can alter any other role (except for system roles such as `mz_system` and `mz_support`). We haven't implemented `ADMIN OPTION`, but maybe we can add the requirement of being a member of the target role.

**Comment by morsapaes** [comment]
> Maybe we can add the requirement of being a member of the target role.

That makes sense to me.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**